src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -137,6 +137,7 @@ public static PodTemplate combine(PodTemplate parent, PodTemplate template) {
 
         PodTemplate podTemplate = new PodTemplate();
         podTemplate.setName(name);
+        podTemplate.setNamespace(!Strings.isNullOrEmpty(template.getNamespace()) ? template.getNamespace() : parent.getNamespace());
```

Comment:
passing a null namespace to the client, result in using whatever is already configured and this this case the one from the cloud definition.